### ISSUE TITLE
Two Users Can Select the Same Square

### ISSUE THREAD

**Issue by wonderchook** [hotosm/osm-tasking-manager#129]
I think this is actually a new issue, because we haven't run into it before.

If two users click on the same square they both are able to check it out. I think they have to do it almost as the same time for it to happen though.

**Comment by ghost** [comment]
This seems to be a limitation of the sqlite database.  What is needed are multi-user locking features such as "select for update" and commit but sqlite does not have these features. More info at http://stackoverflow.com/questions/5102027/can-sqlite-support-multiple-user

**Comment by pgiraud** [comment]
You're probably right.
SQLite has the advantage to be easy to use and backup with its one-file database. I have been thinking about moving to posgreSQL for a new version though.

**Comment by danstowell** [comment]
We also had a similar thing happen to someone last night - two people simultaneously locked the same square and started editing. It wasn't me so I don't have details I'm afraid. This issue has the potential to start arguments!

**Comment by Nick-Tallguy** [comment]
More info on this.
Several months ago (Philippines activation) I was one of two mappers involved in us both mapping the same square. Cannot remember all the details, but it may be one of the instances referred to above.
  In retrospect, the square I was working on was a large complicated one, and I had no problem initially. It may be that I reached the time limit for the square whilst still working on it, and that another user selected it after it was released by Tasking Manager as the time limit had been reached. It was not until we each reached the same area of the square that this created conflicts for us.
I wonder if this could be the case for some of the other instances?
